src/decor.rs

```diff
@@ -8,15 +8,15 @@ pub struct Formatted<T> {
 
 // String representation of a key or a value
 // together with a decoration.
-#[derive(Eq, PartialEq, Clone, Debug, Hash)]
-pub(crate) struct Repr {
+#[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Debug, Hash)]
```

Comment:
I don't think the derived ordering is "correct" here, it will first compare prefix, then suffixes, then raw_value